go-controller/pkg/nbdb/logical_switch_port.go

```diff
@@ -28,6 +28,20 @@ type LogicalSwitchPort struct {
 	Up               *bool             `ovsdb:"up"`
 }
 
+type LSPOption func(lsp *LogicalSwitchPort)
+
+func WithLogicalSwitchPortEnabled(enabled bool) LSPOption {
```

Comment:
The reasoning is different. It goes like this:
- we must set the LSP.Enabled **inside** the `addLogicalPortToNetwork` function where the lsp ops are created (doing it after the ops then appending it is wrong and results with transaction err)
- BUT we don't want the logic of the LSP.Enabled to be inside that function. this is because `addLogicalPortToNetwork` is a `BaseNetworkController` method that runs for all pods and our logic is only relevant to `BaseSecondaryNetworkController`.
- So we should check the desired LSP.Enabled value **before** then pass it to the `addLogicalPortToNetwork` function.
- BUT LSP.Enabled is *bool and we shouldn't pass a boolean to that function.. that's not nice.. 

---> So we introduced the `LSPOption`. 
Yes, it's sort of a builder, but it's a good generalization of what we want to do.

Cons: 
- right now it looks like we're adding a boolean, but in a convoluted way..

Pros: 
- doing so will allow us to move other lsp related logic to the `BaseSecondaryNetworkController` (things like [this](https://github.com/ovn-kubernetes/ovn-kubernetes/blob/master/go-controller/pkg/ovn/base_network_controller_pods.go#L573-L577) which clearly should be relevant only for secondary controller)

Reply:
Honestly, I read this twice and I didn't understand what you're trying to say. I still can't see past the `con` you're listing.